src/wait-for.js

```diff
@@ -160,6 +163,9 @@ function waitFor(
     }
 
     function handleTimeout() {
+      if (finished) {
```

Comment:
Shouldn't we cancel the timeout instead when we finish?

Reply:
When we finish, `onDone` should have been called with `clearTimeout` ([src](https://github.com/testing-library/dom-testing-library/blob/a0bb0050b1ccb7652a60b72d617453400c300837/src/wait-for.js#L113)), therefore I don't really think this one is needed, but it's more for extra-safety.

Reply:
Amended in https://github.com/testing-library/dom-testing-library/pull/1271/commits/6b41483386804687e145119bc1966a2da711eb01